### PR TITLE
Muon Analysis 2: When the instrument changes, a plot should close.

### LINKED ISSUE #26201: MuonAnalysis2 unexpected exception when load run from different instrument
### Expected behavior
Muon Analysis 2 should be able to handle changing between instruments and loading data
### Actual behavior
When trying to load data when another run from a different instroment is already plotted, an error is returned.
### Steps to reproduce the behavior
1. Open Workbench then Muon Analysis 2.
1. Set Instroment to MUSR and load run 62260.
1. Chang instrument to EMU and load run 15098.
1. An unexpected Exception is raised.

Traceback (most recent call last):
  File "C:\MantidNightlyInstall\scripts\Muon\GUI\Common\observer_pattern.py", line 63, in _notify_subscribers_impl
    observer.update(self, self.arg, **self.kwargs)
  File "C:\MantidNightlyInstall\scripts\Muon\GUI\Common\observer_pattern.py", line 89, in update
    self.callback()
  File "C:\MantidNightlyInstall\scripts\Muon\GUI\Common\home_plot_widget\home_plot_widget_presenter.py", line 61, in handle_data_updated
    self.plot_standard_workspaces()
  File "C:\MantidNightlyInstall\scripts\Muon\GUI\Common\home_plot_widget\home_plot_widget_presenter.py", line 109, in plot_standard_workspaces
    self._model.plot(workspace_list, self.get_plot_title())
  File "C:\MantidNightlyInstall\scripts\Muon\GUI\Common\home_plot_widget\home_plot_widget_model.py", line 45, in plot
    self._remove_all_data_workspaces_from_plot()
  File "C:\MantidNightlyInstall\scripts\Muon\GUI\Common\home_plot_widget\home_plot_widget_model.py", line 131, in _remove_all_data_workspaces_from_plot
    self.remove_workpace_from_plot(workspace)
  File "C:\MantidNightlyInstall\scripts\Muon\GUI\Common\home_plot_widget\home_plot_widget_model.py", line 95, in remove_workpace_from_plot
    self.plot_figure.gca().remove_workspace_artists(workspace)
AttributeError: 'AxesSubplot' object has no attribute 'remove_workspace_artists'
### Platforms affected
Windows

### PR DESCRIPTION
**Pull request by Pasarus**
**Description of work.**
There was an issue where loading the data of another instrument caused havoc when updating the plots. So plots now close on instrument change.

**To test:**
- Open Muon Analysis 2 in Workbench
- The instrument should be EMU, load run 15098
- A plot should have appeared
- Change the instrument to any other
- The Plot should close.

Fixes #26201 . 

NEW FEATURE FIX SO NO NEW RELEASE NOTES

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_model.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def remove_workpace_from_plot(self, workspace_name):
         self.plotted_workspaces = [item for item in self.plotted_workspaces if item != workspace_name]
         self.plotted_fit_workspaces = [item for item in self.plotted_fit_workspaces if item != workspace_name]
 
-    def _close_plot(self):
+    def close_plot(self):
```
Comment (Mantid-Matthew) [inline scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_model.py]: This isn't an issue introduced in this PR but seems sensible to address here anyway, can we clear self.plotted_workspaces_inverse_binning and self.plotted_fit_workspaces in close plot when we clear the list self.plotted_workspaces

File: scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_presenter.py [hunk 2]
```diff
@@ -182,3 +182,9 @@ def handle_rebin_options_set(self):
             self._view.set_raw_checkbox_state(False)
         else:
             self._view.set_raw_checkbox_state(True)
+
+    def handle_instrument_changed(self):
+        if self._model.plot_figure is not None:
+            from matplotlib import pyplot as plt
+            plt.close(self._model.plot_figure)
+        self._model.close_plot()
```
Comment (Mantid-Matthew) [inline scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_presenter.py]: I don't think this call is needed as close_plot is automatically linked up to the close signal on the figure window.

**Review by Matthew-Andrew — COMMENTED** [review]
All works for me, a few very minor comments

File: scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_presenter.py (hunk 2 above)
Comment (Pasarus) [reply scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_presenter.py]: it does not work without these lines

**Review by Matthew-Andrew — APPROVED** [review]
Looks good to me

**Pull request merged**